trin-core/src/jsonrpc/handlers.rs

```diff
@@ -33,61 +36,97 @@ impl JsonRpcHandler {
                         Ok(self.discovery.write_with_warn().await.routing_table_info())
                     }
                 },
-                TrinEndpoint::HistoryEndpoint(endpoint) => match self.history_jsonrpc_tx.as_ref() {
-                    Some(tx) => proxy_query_to_history_subnet(tx, endpoint).await,
-                    None => Err("Chain history subnetwork unavailable.".to_string()),
-                },
-                TrinEndpoint::StateEndpoint(endpoint) => match self.state_jsonrpc_tx.as_ref() {
-                    Some(tx) => proxy_query_to_state_subnet(tx, endpoint).await,
-                    None => Err("State subnetwork unavailable.".to_string()),
-                },
-                _ => Err(format!(
+                TrinEndpoint::OverlayEndpoint(endpoint) => {
+                    self.process_overlay_request(endpoint, request.params.clone())
+                        .await
+                }
+                _ => Err(JsonRpcError::InvalidRequest(format!(
                     "Can't process portal network endpoint {:?}",
                     request.endpoint
-                )),
+                ))),
+            };
+            let _ = match response {
+                Ok(val) => request.resp.send(Ok(val)),
+                Err(msg) => request.resp.send(Err(msg.to_string())),
             };
-            let _ = request.resp.send(response);
+        }
+    }
+
+    async fn process_overlay_request(
+        &self,
+        endpoint: OverlayEndpoint,
+        params: Params,
+    ) -> Result<Value, JsonRpcError> {
+        match params.clone() {
+            Params::Array(val) => match &val[0].as_str() {
+                Some("history") => match self.history_jsonrpc_tx.as_ref() {
```

Comment:
Is it worth adding an enum for all possible network identifiers ("state", 'history", .. ) and then matching it here instead of using strings?